### ISSUE TITLE
Add user roles view to UI [3]

### ISSUE THREAD

**Issue by MartinWahnschaffe** [hzi-braunschweig/SORMAS-Project#4462]
### Situation Description
Source: https://github.com/hzi-braunschweig/SORMAS-Project/issues/898#issuecomment-671336289

Depends on #4461 

### Feature Description

Mockup: https://www.sketch.com/s/32266e24-8c84-4006-8e1c-27263b30bdb4/a/xrjMW48

- [x] add view. Should be a tab in the users view
- [x] list all user roles as configured in the database
- jurisdiction should be jurisdiction level
- [x] allow filtering by user right (list all roles that have a certain right) and jurisdiction
- [x] allow filtering by enabled: enabled, disabled, all user roles
- [x] allow detailed export. Each user right should go into a separate column -> can be moved from the users section. See #4882 
- [x] Add USER_ROLE_VIEW right. Give it to the admin default user. For migration give those rights to every user role that is allowed to edit users.

**Comment by kathsievers** [comment]
ISSUE TRACKER COMMENT

**Comment by kwa20** [comment]
ISSUE TRACKER COMMENT

**Comment by vidi42** [comment]
@MartinWahnschaffe is it filtering by archived or by enabled/disabled?

**Comment by MartinWahnschaffe** [comment]
Currently by enabled/disabled, though I'd suggest to refactor this field to "archived" in the context of https://github.com/hzi-braunschweig/SORMAS-Project/issues/4463

**Comment by abrudanancuta** [comment]
Cannot create new user on test , an error pop-up appears. I attached the error that is in the server logs
[NewUserError.txt](https://github.com/hzi-braunschweig/SORMAS-Project/files/9132185/NewUserError.txt)
The same error appears when As Admin select the Users page -> User Roles tab -> New User Role

**Comment by carina29** [comment]
@abrudanancuta the New User Role creation was not requested in this story, I've added the button only because in the mockup from Martin was added. The functionality will work when this story will be merged: https://github.com/hzi-braunschweig/SORMAS-Project/issues/4463

**Comment by ChristopherRiedel** [comment]
> @abrudanancuta the New User Role creation was not requested in this story, I've added the button only because in the mockup from Martin was added. The functionality will work when this story will be merged: #4463
> 
> The User Role page and the export should be tested for this story.

@carina29 But it is not good that we have buttons in the system that are broken. You can never be sure that an issue (#4463) will be finished by the end of the sprint. Therefore all changes that end up in development must be release ready. Please remove or hide the button.

**Comment by abrudanancuta** [comment]
Added a ticket related to User changes https://github.com/hzi-braunschweig/SORMAS-Project/issues/9900

**Comment by abrudanancuta** [comment]
Found 2 Cases for User Management:

**1. Case to Edit User**

As admin logged in go to Users page
Choose a user and open it in Edit mode
Don't change anything just make a simple save

**2. Case to create New User**

As admin logged in go to Users page
Create a new user
Try to save the new user created

Expected Behavior
Edit and Create user should not throw any error

<img width="1440" alt="Screenshot 2022-07-19 at 13 34 44" src="https://user-images.githubusercontent.com/91872179/179954087-0a442fe9-f593-4425-943f-6785d0faceca.png">

**Comment by kwa20** [comment]
I also noticed on test-de.sormas.netzlink.com; 1.75.0-SNAPSHOT (d94204a) that EXTERNAL_LABORATORY is not available for translation on crowdin.

<img width="311" alt="jurisdictionevelTranslation" src="https://user-images.githubusercontent.com/70317594/180173558-feac498e-62e7-4bf4-95fa-b24c19b014d1.PNG">

**Comment by kwa20** [comment]
I tested this feature on test-de2.sormas.netzlink.com (98aa280) and noticed the following things:

1. The USER_ROLE_VIEW right does not seem to have a caption/translation
<img width="441" alt="userRoleViewCaption" src="https://user-images.githubusercontent.com/70317594/182326208-3f1bbf40-81b0-4ffe-b48e-450c0b0867d1.PNG">

2. Sorting the uuid column on the user roles view results in an error
<img width="842" alt="userRoleListUuidSortError" src="https://user-images.githubusercontent.com/70317594/182326352-5e7d6f04-387c-4e15-bb7a-8628a587098b.PNG">

3. The list grid does not use the available space and cuts off text even if the space would allow for the full display of the content.
    a. Create a new user role
    b. enter a long description
    c. check the display on the user roles view
<img width="857" alt="userRolesListResponsive" src="https://user-images.githubusercontent.com/70317594/182326531-854a0c07-73f9-4c1a-b533-0c70f696f5d3.PNG">

**Comment by ChristopherRiedel** [comment]
@kwa20 
> 2. Sorting the uuid column on the user roles view results in an error
Adding this coulmn was't part of this issue -> #4463

**Comment by carina29** [comment]
@kwa20 Thank you for the message and screenshots. I will check your new findings too.

It was not described how should the grid use the space for different columns, so I used some default column widths as it was used for User Management table too ( but there the strings are shorter so is not visible the problem described). :)
I can fix it.

The UUI column as Cristopher said, was not implemented in 4462.

EXTERNAL_LABORATORY is also not a bug related to this story but I've fixed yesterday on my branch.
EXTERNAL_LABORATORY did not have at all a translation added till now but I suppose that it was not displayed in the UI and the bug was not caught.

**Comment by abrudanancuta** [comment]
The export user roles file contains a column with the UUIDs. I would propose to remove that column, could be exposing sensitive data

**Comment by ChristopherRiedel** [comment]
@abrudanancuta 
> The export user roles file contains a column with the UUIDs. I would propose to remove that column, could be exposing sensitive data

We actually always export the uuid of entities as well. Why should this be a problem here and not elsewhere?

**Comment by carina29** [comment]
@abrudanancuta I've asked @MartinWahnschaffe about the Export content ( which columns should be placed in the export and he agreed with the UUID column too).  

@all 
I would suggest that we should separate in the future the new changes ( which was not requested initially in the story description) from the bugs.

Some bugs were discovered in this story( also a part of bugs were not created with the changes of this story but were discovered during the test of this story) and the largest part of the findings were not requested in the story description.

Those new requirements should go into a new story or this old story should have been re estimated since a long time.
We should not change the story requirement after estimation or we should re estimate the story if the requirements are changed . ( In order to finish the task in time and to be able to meet the spring commitment - which should reflect the estimations of the requirements).

**Comment by abrudanancuta** [comment]
New Role is not appearing in the list if has no User rights selected

Steps to reproduce:

1. Create a new Role
2. Select the role to be Enable
3. Save the change
4. Don't select any of the user rights
5. Go back to the list of User Roles
6. Look for the new Role created
7. Go at the User Management Tab
8. Select the Create New User and look for the Role created at step 1

Actual results: In the list, the new Role created cannot be found at any of the status Enable/Disable. But in the Create New User, this new Role created is displayed

Expected results: The new Role created should appear in the list to be able to edit it.

**Comment by abrudanancuta** [comment]
> 3. The list grid does not use the available space and cuts off text even if the space would allow for the full display of the content.
>    a. Create a new user role
>    b. enter a long description
>    c. check the display on the user roles view
> 
> <img alt="userRolesListResponsive" width="857" src="https://user-images.githubusercontent.com/70317594/182326531-854a0c07-73f9-4c1a-b533-0c70f696f5d3.PNG">

for this 3rd point was created a new ticket https://github.com/hzi-braunschweig/SORMAS-Project/issues/10032

**Comment by carina29** [comment]
> New Role is not appearing in the list if has no User rights selected
> 
> Steps to reproduce:
> 
> 1. Create a new Role
> 2. Select the role to be Enable
> 3. Save the change
> 4. Don't select any of the user rights
> 5. Go back to the list of User Roles
> 6. Look for the new Role created
> 7. Go at the User Management Tab
> 8. Select the Create New User and look for the Role created at step 1
> 
> Actual results: In the list, the new Role created cannot be found at any of the status Enable/Disable. But in the Create New User, this new Role created is displayed
> 
> Expected results: The new Role created should appear in the list to be able to edit it.
> 
> **Observation**: The test case is applied also when having the Role with Disable status



This problem will be handled in https://github.com/hzi-braunschweig/SORMAS-Project/issues/10049

**Comment by carina29** [comment]
We agreed with @markusmann-vg that https://github.com/hzi-braunschweig/SORMAS-Project/issues/10032 and https://github.com/hzi-braunschweig/SORMAS-Project/issues/10049 can be fixed under 4462.

I will discard the above mentioned tickets.

**Comment by abrudanancuta** [comment]
Validated on the local environment with the version: 1.75.0-SNAPSHOT (6c89d83)
